hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonRecordReader.java

```diff
@@ -78,7 +80,13 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext context)
     readSupport.initialize(queryModel.getProjectionColumns(),
         queryModel.getAbsoluteTableIdentifier());
     try {
-      carbonIterator = new ChunkRowIterator(queryExecutor.execute(queryModel));
+      if (queryModel.isVectorReader()) {
+        carbonIterator = new VectorChunkRowIterator(
```

Comment:
I moved out this logic out of the class. And new `VectorizedCarbonRecordReader` is created.